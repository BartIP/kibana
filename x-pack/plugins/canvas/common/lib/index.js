/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License;
 * you may not use this file except in compliance with the Elastic License.
 */

export * from './datatable';
<<<<<<< HEAD
export * from './arg';
export * from './ast';
=======
>>>>>>> ff49a1c6742d67fa5daed569ff3bb269783f6bd1
export * from './autocomplete';
export * from './constants';
export * from './dataurl';
export * from './errors';
export * from './expression_form_handlers';
export * from './fetch';
export * from './find_in_object';
<<<<<<< HEAD
export * from './fn';
export * from './fonts';
export * from './functions_registry';
export * from './get_by_alias';
export * from './get_colors_from_palette';
export * from './get_field_type';
export * from './get_legend_config';
export * from './get_type';
export * from './grammar';
=======
export * from './fonts';
export * from './get_colors_from_palette';
export * from './get_field_type';
export * from './get_legend_config';
>>>>>>> ff49a1c6742d67fa5daed569ff3bb269783f6bd1
export * from './handlebars';
export * from './hex_to_rgb';
export * from './httpurl';
export * from './latest_change';
export * from './missing_asset';
export * from './palettes';
export * from './pivot_object_array';
<<<<<<< HEAD
export * from './registry';
export * from './resolve_dataurl';
export * from './serialize';
export * from './type';
export * from './types_registry';
=======
export * from './resolve_dataurl';
>>>>>>> ff49a1c6742d67fa5daed569ff3bb269783f6bd1
export * from './unquote_string';
export * from './url';
