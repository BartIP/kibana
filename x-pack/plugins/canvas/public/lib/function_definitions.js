/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License;
 * you may not use this file except in compliance with the Elastic License.
 */

import uniqBy from 'lodash.uniqby';
<<<<<<< HEAD
import { getServerFunctions } from '../state/selectors/app';
import { getBrowserRegistries } from './browser_registries';
=======
import { getBrowserRegistries } from '@kbn/interpreter/public/browser_registries';
import { getServerFunctions } from '../state/selectors/app';
>>>>>>> ff49a1c6742d67fa5daed569ff3bb269783f6bd1

export async function getFunctionDefinitions(state) {
  const { browserFunctions } = await getBrowserRegistries();
  const serverFunctions = getServerFunctions(state);
  return uniqBy(serverFunctions.concat(browserFunctions.toArray()), 'name');
}
