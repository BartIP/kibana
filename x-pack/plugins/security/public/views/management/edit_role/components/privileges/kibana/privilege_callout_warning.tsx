/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License;
 * you may not use this file except in compliance with the Elastic License.
 */
import { EuiCallOut } from '@elastic/eui';
<<<<<<< HEAD
=======
import { FormattedMessage } from '@kbn/i18n/react';
>>>>>>> ff49a1c6742d67fa5daed569ff3bb269783f6bd1
import React, { Component } from 'react';
import { KibanaPrivilege } from '../../../../../../../common/model/kibana_privilege';
import { NO_PRIVILEGE_VALUE } from '../../../lib/constants';

interface Props {
  basePrivilege: KibanaPrivilege;
  isReservedRole: boolean;
}

interface State {
  showImpactedSpaces: boolean;
}

export class PrivilegeCalloutWarning extends Component<Props, State> {
  public state = {
    showImpactedSpaces: false,
  };

  public render() {
    const { basePrivilege, isReservedRole } = this.props;

    let callout = null;

    if (basePrivilege === 'all') {
      if (isReservedRole) {
        callout = (
          <EuiCallOut
            color="warning"
            iconType="iInCircle"
<<<<<<< HEAD
            title={"Cannot customize a reserved role's space privileges"}
          >
            <p>
              This role always grants full access to all spaces. To customize privileges for
              individual spaces, you must create a new role.
=======
            title={
              <FormattedMessage
                id="xpack.security.management.editRoles.privilegeCalloutWarning.notPossibleToCustomizeReservedRoleSpacePrivilegesTitle"
                defaultMessage="Cannot customize a reserved role's space privileges"
              />
            }
          >
            <p>
              <FormattedMessage
                id="xpack.security.management.editRoles.privilegeCalloutWarning.howToCustomizePrivilegesDescription"
                defaultMessage="This role always grants full access to all spaces. To customize privileges for
                individual spaces, you must create a new role."
              />
>>>>>>> ff49a1c6742d67fa5daed569ff3bb269783f6bd1
            </p>
          </EuiCallOut>
        );
      } else {
        callout = (
          <EuiCallOut
            color="warning"
            iconType="iInCircle"
<<<<<<< HEAD
            title={'Minimum privilege is too high to customize individual spaces'}
          >
            <p>
              Setting the minimum privilege to <strong>all</strong> grants full access to all
              spaces. To customize privileges for individual spaces, the minimum privilege must be
              either <strong>read</strong> or <strong>none</strong>.
=======
            title={
              <FormattedMessage
                id="xpack.security.management.editRoles.privilegeCalloutWarning.minimumPrivilegeTitle"
                defaultMessage="Minimum privilege is too high to customize individual spaces"
              />
            }
          >
            <p>
              <FormattedMessage
                id="xpack.security.management.editRoles.privilegeCalloutWarning.howToCustomizePrivilegesForIndividualSpacesDescription"
                defaultMessage="Setting the minimum privilege to {allText} grants full access to all
                spaces. To customize privileges for individual spaces, the minimum privilege must be
                either {readText} or {noneText}."
                values={{
                  allText: (
                    <strong>
                      <FormattedMessage
                        id="xpack.security.management.editRoles.privilegeCalloutWarning.allText"
                        defaultMessage="all"
                      />
                    </strong>
                  ),
                  readText: (
                    <strong>
                      <FormattedMessage
                        id="xpack.security.management.editRoles.privilegeCalloutWarning.readText"
                        defaultMessage="read"
                      />
                    </strong>
                  ),
                  noneText: (
                    <strong>
                      <FormattedMessage
                        id="xpack.security.management.editRoles.privilegeCalloutWarning.noneText"
                        defaultMessage="none"
                      />
                    </strong>
                  ),
                }}
              />
>>>>>>> ff49a1c6742d67fa5daed569ff3bb269783f6bd1
            </p>
          </EuiCallOut>
        );
      }
    }

    if (basePrivilege === 'read') {
      if (isReservedRole) {
        callout = (
          <EuiCallOut
            color="warning"
            iconType="iInCircle"
<<<<<<< HEAD
            title={"Cannot customize a reserved role's space privileges"}
          >
            <p>
              This role always grants read access to all spaces. To customize privileges for
              individual spaces, you must create a new role.
=======
            title={
              <FormattedMessage
                id="xpack.security.management.editRoles.privilegeCalloutWarning.notPossibleToCustomizeReservedRoleSpacePrivilegesTitle"
                defaultMessage="Cannot customize a reserved role's space privileges"
              />
            }
          >
            <p>
              <FormattedMessage
                id="xpack.security.management.editRoles.privilegeCalloutWarning.alwaysGrantReadAccessToAllSpacesTitle"
                defaultMessage="This role always grants read access to all spaces. To customize privileges for
                individual spaces, you must create a new role."
              />
>>>>>>> ff49a1c6742d67fa5daed569ff3bb269783f6bd1
            </p>
          </EuiCallOut>
        );
      } else {
        callout = (
          <EuiCallOut
            color="primary"
            iconType="iInCircle"
            title={
              <span>
<<<<<<< HEAD
                The minimal possible privilege is <strong>read</strong>.
=======
                <FormattedMessage
                  id="xpack.security.management.editRoles.privilegeCalloutWarning.minimalPossiblePrivilageTitle"
                  defaultMessage="The minimal possible privilege is {readText}."
                  values={{
                    readText: (
                      <strong>
                        <FormattedMessage
                          id="xpack.security.management.editRoles.privilegeCalloutWarning.readText"
                          defaultMessage="read"
                        />
                      </strong>
                    ),
                  }}
                />
>>>>>>> ff49a1c6742d67fa5daed569ff3bb269783f6bd1
              </span>
            }
          />
        );
      }
    }

    if (basePrivilege === NO_PRIVILEGE_VALUE && isReservedRole) {
      callout = (
        <EuiCallOut
          color="warning"
          iconType="iInCircle"
<<<<<<< HEAD
          title={"Cannot customize a reserved role's space privileges"}
        >
          <p>
            This role never grants access to any spaces within Kibana. To customize privileges for
            individual spaces, you must create a new role.
=======
          title={
            <FormattedMessage
              id="xpack.security.management.editRoles.privilegeCalloutWarning.notPossibleToCustomizeReservedRoleSpacePrivilegesTitle"
              defaultMessage="Cannot customize a reserved role's space privileges"
            />
          }
        >
          <p>
            <FormattedMessage
              id="xpack.security.management.editRoles.privilegeCalloutWarning.neverGrantReadAccessToAllSpacesTitle"
              defaultMessage="This role never grants access to any spaces within Kibana. To customize privileges for
              individual spaces, you must create a new role."
            />
>>>>>>> ff49a1c6742d67fa5daed569ff3bb269783f6bd1
          </p>
        </EuiCallOut>
      );
    }

    return callout;
  }
}
