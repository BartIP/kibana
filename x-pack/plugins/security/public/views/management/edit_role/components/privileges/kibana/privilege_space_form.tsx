/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License;
 * you may not use this file except in compliance with the Elastic License.
 */

import { EuiButtonIcon, EuiFlexGroup, EuiFlexItem, EuiFormRow } from '@elastic/eui';
<<<<<<< HEAD
=======
import { FormattedMessage } from '@kbn/i18n/react';
>>>>>>> ff49a1c6742d67fa5daed569ff3bb269783f6bd1
import React, { Component } from 'react';
import { Space } from '../../../../../../../../spaces/common/model/space';
import { KibanaPrivilege } from '../../../../../../../common/model/kibana_privilege';
import { RoleValidator } from '../../../lib/validate_role';
import { PrivilegeSelector } from './privilege_selector';
import { SpaceSelector } from './space_selector';

interface Props {
  availableSpaces: Space[];
  selectedSpaceIds: string[];
  availablePrivileges: KibanaPrivilege[];
  selectedPrivilege: KibanaPrivilege | null;
  onChange: (
    params: {
      spaces: string[];
      privilege: KibanaPrivilege | null;
    }
  ) => void;
  onDelete: () => void;
  validator: RoleValidator;
}

export class PrivilegeSpaceForm extends Component<Props, {}> {
  public render() {
    const {
      availableSpaces,
      selectedSpaceIds,
      availablePrivileges,
      selectedPrivilege,
      validator,
    } = this.props;

    return (
      <EuiFlexGroup responsive={false}>
        <EuiFlexItem>
          <EuiFormRow
<<<<<<< HEAD
            label={'Spaces'}
=======
            label={
              <FormattedMessage
                id="xpack.security.management.editRoles.privilegeSpaceForm.spacesFormRowLabel"
                defaultMessage="Spaces"
              />
            }
>>>>>>> ff49a1c6742d67fa5daed569ff3bb269783f6bd1
            {...validator.validateSelectedSpaces(selectedSpaceIds, selectedPrivilege)}
          >
            <SpaceSelector
              spaces={availableSpaces}
              selectedSpaceIds={selectedSpaceIds}
              onChange={this.onSelectedSpacesChange}
            />
          </EuiFormRow>
        </EuiFlexItem>
        <EuiFlexItem>
          <EuiFormRow
<<<<<<< HEAD
            label={'Privilege'}
=======
            label={
              <FormattedMessage
                id="xpack.security.management.editRoles.privilegeSpaceForm.privilegeFormRowLabel"
                defaultMessage="Privilege"
              />
            }
>>>>>>> ff49a1c6742d67fa5daed569ff3bb269783f6bd1
            {...validator.validateSelectedPrivilege(selectedSpaceIds, selectedPrivilege)}
          >
            <PrivilegeSelector
              data-test-subj={'space-form-priv-selector'}
              availablePrivileges={availablePrivileges}
              value={selectedPrivilege}
              onChange={this.onPrivilegeChange}
            />
          </EuiFormRow>
        </EuiFlexItem>
        <EuiFlexItem grow={false}>
          <EuiFormRow hasEmptyLabelSpace>
            <EuiButtonIcon
              aria-label={'Delete space privilege'}
              color={'danger'}
              onClick={this.props.onDelete}
              iconType={'trash'}
            />
          </EuiFormRow>
        </EuiFlexItem>
      </EuiFlexGroup>
    );
  }

  public onSelectedSpacesChange = (selectedSpaceIds: string[]) => {
    this.props.onChange({
      spaces: selectedSpaceIds,
      privilege: this.props.selectedPrivilege,
    });
  };

  public onPrivilegeChange = (privilege: KibanaPrivilege) => {
    this.props.onChange({
      spaces: this.props.selectedSpaceIds,
      privilege,
    });
  };
}
