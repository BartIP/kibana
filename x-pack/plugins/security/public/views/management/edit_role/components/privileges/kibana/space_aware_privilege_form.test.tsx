/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License;
 * you may not use this file except in compliance with the Elastic License.
 */

<<<<<<< HEAD
import { mount, shallow } from 'enzyme';
import React from 'react';
=======
import React from 'react';
import { mountWithIntl, shallowWithIntl } from 'test_utils/enzyme_helpers';
>>>>>>> ff49a1c6742d67fa5daed569ff3bb269783f6bd1
import { RoleValidator } from '../../../lib/validate_role';
import { PrivilegeCalloutWarning } from './privilege_callout_warning';
import { PrivilegeSpaceForm } from './privilege_space_form';
import { PrivilegeSpaceTable } from './privilege_space_table';
import { SpaceAwarePrivilegeForm } from './space_aware_privilege_form';

const buildProps = (customProps: any = {}) => {
  return {
    role: {
      name: '',
      elasticsearch: {
        cluster: ['manage'],
        indices: [],
        run_as: [],
      },
      kibana: {
        global: [],
        space: {},
      },
    },
    spaces: [
      {
        id: 'default',
        name: 'Default Space',
        _reserved: true,
      },
      {
        id: 'marketing',
        name: 'Marketing',
      },
    ],
    userProfile: { hasCapability: () => true },
    editable: true,
    kibanaAppPrivileges: ['all', 'read'],
    onChange: jest.fn(),
    validator: new RoleValidator(),
    ...customProps,
  };
};

describe('<SpaceAwarePrivilegeForm>', () => {
  it('renders without crashing', () => {
<<<<<<< HEAD
    expect(shallow(<SpaceAwarePrivilegeForm {...buildProps()} />)).toMatchSnapshot();
=======
    expect(
      shallowWithIntl(<SpaceAwarePrivilegeForm.WrappedComponent {...buildProps()} />)
    ).toMatchSnapshot();
>>>>>>> ff49a1c6742d67fa5daed569ff3bb269783f6bd1
  });

  it('shows the space table if exisitng space privileges are declared', () => {
    const props = buildProps({
      role: {
        elasticsearch: {
          cluster: ['manage'],
        },
        kibana: {
          global: ['read'],
          space: {
            default: ['all'],
          },
        },
      },
    });

<<<<<<< HEAD
    const wrapper = mount(<SpaceAwarePrivilegeForm {...props} />);
=======
    const wrapper = mountWithIntl(<SpaceAwarePrivilegeForm.WrappedComponent {...props} />);
>>>>>>> ff49a1c6742d67fa5daed569ff3bb269783f6bd1

    const table = wrapper.find(PrivilegeSpaceTable);
    expect(table).toHaveLength(1);
  });

  it('hides the space table if there are no existing space privileges', () => {
    const props = buildProps();

<<<<<<< HEAD
    const wrapper = mount(<SpaceAwarePrivilegeForm {...props} />);
=======
    const wrapper = mountWithIntl(<SpaceAwarePrivilegeForm.WrappedComponent {...props} />);
>>>>>>> ff49a1c6742d67fa5daed569ff3bb269783f6bd1

    const table = wrapper.find(PrivilegeSpaceTable);
    expect(table).toMatchSnapshot();
  });

  it('adds a form row when clicking the "Add Space Privilege" button', () => {
    const props = buildProps({
      role: {
        elasticsearch: {
          cluster: ['manage'],
        },
        kibana: {
          global: ['read'],
          space: {
            default: ['all'],
          },
        },
      },
    });

<<<<<<< HEAD
    const wrapper = mount(<SpaceAwarePrivilegeForm {...props} />);
=======
    const wrapper = mountWithIntl(<SpaceAwarePrivilegeForm.WrappedComponent {...props} />);
>>>>>>> ff49a1c6742d67fa5daed569ff3bb269783f6bd1
    expect(wrapper.find(PrivilegeSpaceForm)).toHaveLength(0);

    wrapper.find('button[data-test-subj="addSpacePrivilegeButton"]').simulate('click');

    expect(wrapper.find(PrivilegeSpaceForm)).toHaveLength(1);
  });

  describe('with minimum privilege set to "all"', () => {
    it('does not allow space privileges to be customized', () => {
      const props = buildProps({
        role: {
          elasticsearch: {
            cluster: ['manage'],
          },
          kibana: {
            global: ['all'],
            space: {
              default: ['all'],
            },
          },
        },
      });

<<<<<<< HEAD
      const wrapper = mount(<SpaceAwarePrivilegeForm {...props} />);
=======
      const wrapper = mountWithIntl(<SpaceAwarePrivilegeForm.WrappedComponent {...props} />);
>>>>>>> ff49a1c6742d67fa5daed569ff3bb269783f6bd1

      const warning = wrapper.find(PrivilegeCalloutWarning);
      expect(warning.props()).toMatchObject({
        basePrivilege: 'all',
      });

      const table = wrapper.find(PrivilegeSpaceTable);
      expect(table).toHaveLength(0);

      const addPrivilegeButton = wrapper.find('[data-test-subj="addSpacePrivilegeButton"]');
      expect(addPrivilegeButton).toHaveLength(0);
    });
  });

  describe('with minimum privilege set to "read"', () => {
    it('shows a warning about minimum privilege', () => {
      const props = buildProps({
        role: {
          elasticsearch: {
            cluster: ['manage'],
          },
          kibana: {
            global: ['read'],
            space: {
              default: ['all'],
            },
          },
        },
      });

<<<<<<< HEAD
      const wrapper = mount(<SpaceAwarePrivilegeForm {...props} />);
=======
      const wrapper = mountWithIntl(<SpaceAwarePrivilegeForm.WrappedComponent {...props} />);
>>>>>>> ff49a1c6742d67fa5daed569ff3bb269783f6bd1

      const warning = wrapper.find(PrivilegeCalloutWarning);
      expect(warning.props()).toMatchObject({
        basePrivilege: 'read',
      });
    });

    it('allows space privileges to be customized', () => {
      const props = buildProps({
        role: {
          elasticsearch: {
            cluster: ['manage'],
          },
          kibana: {
            global: ['read'],
            space: {
              default: ['all'],
            },
          },
        },
      });

<<<<<<< HEAD
      const wrapper = mount(<SpaceAwarePrivilegeForm {...props} />);
=======
      const wrapper = mountWithIntl(<SpaceAwarePrivilegeForm.WrappedComponent {...props} />);
>>>>>>> ff49a1c6742d67fa5daed569ff3bb269783f6bd1

      const table = wrapper.find(PrivilegeSpaceTable);
      expect(table).toHaveLength(1);

      const addPrivilegeButton = wrapper.find('button[data-test-subj="addSpacePrivilegeButton"]');
      expect(addPrivilegeButton).toHaveLength(1);
    });
  });

  describe('with minimum privilege set to "none"', () => {
    it('does not show a warning about minimum privilege', () => {
      const props = buildProps({
        role: {
          elasticsearch: {
            cluster: ['manage'],
          },
          kibana: {
            global: [],
            space: {
              default: ['all'],
            },
          },
        },
      });

<<<<<<< HEAD
      const wrapper = mount(<SpaceAwarePrivilegeForm {...props} />);
=======
      const wrapper = mountWithIntl(<SpaceAwarePrivilegeForm.WrappedComponent {...props} />);
>>>>>>> ff49a1c6742d67fa5daed569ff3bb269783f6bd1

      const warning = wrapper.find(PrivilegeCalloutWarning);
      expect(warning).toHaveLength(0);
    });

    it('allows space privileges to be customized', () => {
      const props = buildProps({
        role: {
          elasticsearch: {
            cluster: ['manage'],
          },
          kibana: {
            global: [],
            space: {
              default: ['all'],
            },
          },
        },
      });

<<<<<<< HEAD
      const wrapper = mount(<SpaceAwarePrivilegeForm {...props} />);
=======
      const wrapper = mountWithIntl(<SpaceAwarePrivilegeForm.WrappedComponent {...props} />);
>>>>>>> ff49a1c6742d67fa5daed569ff3bb269783f6bd1

      const table = wrapper.find(PrivilegeSpaceTable);
      expect(table).toHaveLength(1);

      const addPrivilegeButton = wrapper.find('button[data-test-subj="addSpacePrivilegeButton"]');
      expect(addPrivilegeButton).toHaveLength(1);
    });
  });

  describe('with user profile disabling "manageSpaces"', () => {
    it('renders a warning message instead of the privilege form', () => {
      const props = buildProps({
        userProfile: {
          hasCapability: (capability: string) => {
            if (capability === 'manageSpaces') {
              return false;
            }
            throw new Error(`unexpected call to hasCapability: ${capability}`);
          },
        },
      });

<<<<<<< HEAD
      const wrapper = shallow(<SpaceAwarePrivilegeForm {...props} />);
=======
      const wrapper = shallowWithIntl(<SpaceAwarePrivilegeForm.WrappedComponent {...props} />);
>>>>>>> ff49a1c6742d67fa5daed569ff3bb269783f6bd1
      expect(wrapper).toMatchSnapshot();
    });
  });
});
