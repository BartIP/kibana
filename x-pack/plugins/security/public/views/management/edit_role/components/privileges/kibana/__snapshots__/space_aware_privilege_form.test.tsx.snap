// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SpaceAwarePrivilegeForm> hides the space table if there are no existing space privileges 1`] = `
<<<<<<< HEAD
<PrivilegeSpaceTable
=======
<InjectIntl(PrivilegeSpaceTableUI)
>>>>>>> ff49a1c6742d67fa5daed569ff3bb269783f6bd1
  availablePrivileges={
    Array [
      "all",
      "read",
    ]
  }
  onChange={[Function]}
  role={
    Object {
      "elasticsearch": Object {
        "cluster": Array [
          "manage",
        ],
        "indices": Array [],
        "run_as": Array [],
      },
      "kibana": Object {
        "global": Array [],
        "space": Object {},
      },
      "name": "",
    }
  }
  spacePrivileges={Object {}}
  spaces={
    Array [
      Object {
        "_reserved": true,
        "id": "default",
        "name": "Default Space",
      },
      Object {
        "id": "marketing",
        "name": "Marketing",
      },
    ]
  }
<<<<<<< HEAD
/>
=======
>
  <PrivilegeSpaceTableUI
    availablePrivileges={
      Array [
        "all",
        "read",
      ]
    }
    intl={
      Object {
        "defaultFormats": Object {
          "date": Object {
            "full": Object {
              "day": "numeric",
              "month": "long",
              "weekday": "long",
              "year": "numeric",
            },
            "long": Object {
              "day": "numeric",
              "month": "long",
              "year": "numeric",
            },
            "medium": Object {
              "day": "numeric",
              "month": "short",
              "year": "numeric",
            },
            "short": Object {
              "day": "numeric",
              "month": "numeric",
              "year": "2-digit",
            },
          },
          "number": Object {
            "currency": Object {
              "style": "currency",
            },
            "percent": Object {
              "style": "percent",
            },
          },
          "time": Object {
            "full": Object {
              "hour": "numeric",
              "minute": "numeric",
              "second": "numeric",
              "timeZoneName": "short",
            },
            "long": Object {
              "hour": "numeric",
              "minute": "numeric",
              "second": "numeric",
              "timeZoneName": "short",
            },
            "medium": Object {
              "hour": "numeric",
              "minute": "numeric",
              "second": "numeric",
            },
            "short": Object {
              "hour": "numeric",
              "minute": "numeric",
            },
          },
        },
        "defaultLocale": "en",
        "formatDate": [Function],
        "formatHTMLMessage": [Function],
        "formatMessage": [Function],
        "formatNumber": [Function],
        "formatPlural": [Function],
        "formatRelative": [Function],
        "formatTime": [Function],
        "formats": Object {},
        "formatters": Object {
          "getDateTimeFormat": [Function],
          "getMessageFormat": [Function],
          "getNumberFormat": [Function],
          "getPluralFormat": [Function],
          "getRelativeFormat": [Function],
        },
        "locale": "en",
        "messages": Object {},
        "now": [Function],
        "onError": [Function],
        "textComponent": Symbol(react.fragment),
        "timeZone": null,
      }
    }
    onChange={[Function]}
    role={
      Object {
        "elasticsearch": Object {
          "cluster": Array [
            "manage",
          ],
          "indices": Array [],
          "run_as": Array [],
        },
        "kibana": Object {
          "global": Array [],
          "space": Object {},
        },
        "name": "",
      }
    }
    spacePrivileges={Object {}}
    spaces={
      Array [
        Object {
          "_reserved": true,
          "id": "default",
          "name": "Default Space",
        },
        Object {
          "id": "marketing",
          "name": "Marketing",
        },
      ]
    }
  />
</InjectIntl(PrivilegeSpaceTableUI)>
>>>>>>> ff49a1c6742d67fa5daed569ff3bb269783f6bd1
`;

exports[`<SpaceAwarePrivilegeForm> renders without crashing 1`] = `
<React.Fragment>
  <EuiDescribedFormGroup
    description={
      <p>
<<<<<<< HEAD
        Specify the minimum actions users can perform in your spaces.
=======
        <FormattedMessage
          defaultMessage="Specify the minimum actions users can perform in your spaces."
          id="xpack.security.management.editRoles.spaceAwarePrivilegeForm.minimumActionsUserCanPerformInYourSpacesDescription"
          values={Object {}}
        />
>>>>>>> ff49a1c6742d67fa5daed569ff3bb269783f6bd1
      </p>
    }
    fullWidth={false}
    gutterSize="l"
    title={
      <h3>
<<<<<<< HEAD
        Minimum privileges for all spaces
=======
        <FormattedMessage
          defaultMessage="Minimum privileges for all spaces"
          id="xpack.security.management.editRoles.spaceAwarePrivilegeForm.minPrivilegesForAllSpacesTitle"
          values={Object {}}
        />
>>>>>>> ff49a1c6742d67fa5daed569ff3bb269783f6bd1
      </h3>
    }
    titleSize="xs"
  >
    <EuiFormRow
      describedByIds={Array []}
      fullWidth={false}
      hasEmptyLabelSpace={true}
      helpText="No access to spaces"
    >
      <PrivilegeSelector
        allowNone={true}
        availablePrivileges={
          Array [
            "all",
            "read",
          ]
        }
        data-test-subj="kibanaMinimumPrivilege"
        disabled={false}
        onChange={[Function]}
        value="none"
      />
    </EuiFormRow>
  </EuiDescribedFormGroup>
  <EuiSpacer
    size="l"
  />
  <React.Fragment>
    <EuiTitle
      size="xs"
      textTransform="none"
    >
      <h3>
<<<<<<< HEAD
        Higher privileges for individual spaces
=======
        <FormattedMessage
          defaultMessage="Higher privileges for individual spaces"
          id="xpack.security.management.editRoles.spaceAwarePrivilegeForm.higherPrivilegesForIndividualSpacesTitle"
          values={Object {}}
        />
>>>>>>> ff49a1c6742d67fa5daed569ff3bb269783f6bd1
      </h3>
    </EuiTitle>
    <EuiSpacer
      size="s"
    />
    <EuiText
      color="subdued"
      grow={false}
      size="s"
    >
      <p>
<<<<<<< HEAD
        Grant more privileges on a per space basis. For example, if the privileges are
         
        <strong>
          read
        </strong>
         for all spaces, you can set the privileges to 
        <strong>
          all
        </strong>
         
        for an individual space.
=======
        <FormattedMessage
          defaultMessage="Grant more privileges on a per space basis. For example, if the privileges are
              {read} for all spaces, you can set the privileges to {all}
              for an individual space."
          id="xpack.security.management.editRoles.spaceAwarePrivilegeForm.grantMorePrivilegesTitle"
          values={
            Object {
              "all": <strong>
                <FormattedMessage
                  defaultMessage="all"
                  id="xpack.security.management.editRoles.spaceAwarePrivilegeForm.allText"
                  values={Object {}}
                />
              </strong>,
              "read": <strong>
                <FormattedMessage
                  defaultMessage="read"
                  id="xpack.security.management.editRoles.spaceAwarePrivilegeForm.readText"
                  values={Object {}}
                />
              </strong>,
            }
          }
        />
>>>>>>> ff49a1c6742d67fa5daed569ff3bb269783f6bd1
      </p>
    </EuiText>
    <EuiSpacer
      size="s"
    />
    <React.Fragment>
<<<<<<< HEAD
      <PrivilegeSpaceTable
=======
      <InjectIntl(PrivilegeSpaceTableUI)
>>>>>>> ff49a1c6742d67fa5daed569ff3bb269783f6bd1
        availablePrivileges={
          Array [
            "all",
            "read",
          ]
        }
        onChange={[Function]}
        role={
          Object {
            "elasticsearch": Object {
              "cluster": Array [
                "manage",
              ],
              "indices": Array [],
              "run_as": Array [],
            },
            "kibana": Object {
              "global": Array [],
              "space": Object {},
            },
            "name": "",
          }
        }
        spacePrivileges={Object {}}
        spaces={
          Array [
            Object {
              "_reserved": true,
              "id": "default",
              "name": "Default Space",
            },
            Object {
              "id": "marketing",
              "name": "Marketing",
            },
          ]
        }
      />
    </React.Fragment>
    <EuiFlexGroup
      alignItems="baseline"
      component="div"
      direction="row"
      gutterSize="l"
      justifyContent="flexStart"
      responsive={true}
      wrap={false}
    >
      <EuiFlexItem
        component="div"
        grow={false}
      >
        <EuiButton
          color="primary"
          data-test-subj="addSpacePrivilegeButton"
          fill={false}
          iconSide="left"
          iconType="plusInCircle"
          onClick={[Function]}
          size="s"
          type="button"
        >
<<<<<<< HEAD
          Add space privilege
=======
          <FormattedMessage
            defaultMessage="Add space privilege"
            id="xpack.security.management.editRoles.spaceAwarePrivilegeForm.addSpacePrivilegeTitle"
            values={Object {}}
          />
>>>>>>> ff49a1c6742d67fa5daed569ff3bb269783f6bd1
        </EuiButton>
      </EuiFlexItem>
      <EuiFlexItem
        component="div"
        grow={true}
      >
<<<<<<< HEAD
        <ImpactedSpacesFlyout
=======
        <InjectIntl(ImpactedSpacesFlyoutUI)
>>>>>>> ff49a1c6742d67fa5daed569ff3bb269783f6bd1
          role={
            Object {
              "elasticsearch": Object {
                "cluster": Array [
                  "manage",
                ],
                "indices": Array [],
                "run_as": Array [],
              },
              "kibana": Object {
                "global": Array [],
                "space": Object {},
              },
              "name": "",
            }
          }
          spaces={
            Array [
              Object {
                "_reserved": true,
                "id": "default",
                "name": "Default Space",
              },
              Object {
                "id": "marketing",
                "name": "Marketing",
              },
            ]
          }
          userProfile={
            Object {
              "hasCapability": [Function],
            }
          }
        />
      </EuiFlexItem>
    </EuiFlexGroup>
  </React.Fragment>
</React.Fragment>
`;

exports[`<SpaceAwarePrivilegeForm> with user profile disabling "manageSpaces" renders a warning message instead of the privilege form 1`] = `
<EuiCallOut
  color="danger"
  iconType="alert"
  size="m"
  title={
    <p>
<<<<<<< HEAD
      Insufficient Privileges
=======
      <FormattedMessage
        defaultMessage="Insufficient Privileges"
        id="xpack.security.management.editRoles.spaceAwarePrivilegeForm.insufficientPrivilegesDescription"
        values={Object {}}
      />
>>>>>>> ff49a1c6742d67fa5daed569ff3bb269783f6bd1
    </p>
  }
>
  <p>
<<<<<<< HEAD
    You are not authorized to view all available spaces.
  </p>
  <p>
    Please ensure your account has all privileges granted by the
     
    <strong>
      kibana_user
    </strong>
     role, and try again.
=======
    <FormattedMessage
      defaultMessage="You are not authorized to view all available spaces."
      id="xpack.security.management.editRoles.spaceAwarePrivilegeForm.howToViewAllAvailableSpacesDescription"
      values={Object {}}
    />
  </p>
  <p>
    <FormattedMessage
      defaultMessage="Please ensure your account has all privileges granted by the
              {kibanaUser} role, and try again."
      id="xpack.security.management.editRoles.spaceAwarePrivilegeForm.ensureAccountHasAllPrivilegesGrantedDescription"
      values={
        Object {
          "kibanaUser": <strong>
            <FormattedMessage
              defaultMessage="kibana_user"
              id="xpack.security.management.editRoles.spaceAwarePrivilegeForm.kibanaUserTitle"
              values={Object {}}
            />
          </strong>,
        }
      }
    />
>>>>>>> ff49a1c6742d67fa5daed569ff3bb269783f6bd1
  </p>
</EuiCallOut>
`;
